src/components/item/sharing/ConfirmMembership.tsx

```diff
@@ -0,0 +1,99 @@
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
+
+import { ItemMembership, PermissionLevel } from '@graasp/sdk';
+import { ItemRecord } from '@graasp/sdk/frontend';
+import { Button } from '@graasp/ui';
+
+import { useCurrentUserContext } from '@/components/context/CurrentUserContext';
+
+import { useBuilderTranslation } from '../../../config/i18n';
+import { mutations } from '../../../config/queryClient';
+import { CONFIRM_MEMBERSHIP_DELETE_BUTTON_ID } from '../../../config/selectors';
+import { BUILDER } from '../../../langs/constants';
+import CancelButton from '../../common/CancelButton';
+
+const labelId = 'alert-dialog-title';
+const descriptionId = 'alert-dialog-description';
+
+type Props = {
+  open?: boolean;
+  handleClose: () => void;
+  item: ItemRecord;
+  memberToDelete: ItemMembership | null;
+  hasOnlyOneAdmin: boolean;
+};
+
+const DeleteItemDialog = ({
+  item,
+  open = false,
+  handleClose,
+  memberToDelete,
+  hasOnlyOneAdmin = false,
+}: Props): JSX.Element => {
+  const { t: translateBuilder } = useBuilderTranslation();
+  const { data: member } = useCurrentUserContext();
+
+  const { mutate: deleteItemMembership } = mutations.useDeleteItemMembership();
+
+  const onDelete = () => {
+    if (memberToDelete?.id) {
+      deleteItemMembership({ id: memberToDelete?.id, itemId: item.id });
+      handleClose();
+    }
+  };
+
+  let dialogText = '';
+  // incase of deleting the only admin
+  if (
+    hasOnlyOneAdmin &&
+    memberToDelete?.permission === PermissionLevel?.Admin
```

Comment:
See my comment (above). Don't use `?.` everywhere. Especially `PermissionLevel?.Admin` does not make sense.